### PR TITLE
Install only production dependencies in shell apps

### PR DESCRIPTION
**Pull request by sjchmiela**
### Checklist
- [x] I've read the [Contribution Guidelines](https://github.com/expo/turtle/blob/master/CONTRIBUTING.md).
- [x] I've updated the [CHANGELOG](https://github.com/expo/turtle/blob/master/CHANGELOG.md) if necessary.
- [x] I've ensured the unit and smoke tests are still passing - either by running `yarn test:unit` and `yarn test:smoke:[ios|android]` or by checking the appropriate CircleCI builds' statuses.
- [x] **I've manually tested whether the changes I made work as expected.**

### Motivation and Context

The build process doesn't use any of the `devDependencies` of shell apps and installing them makes the process slower and using more memory.

### Description

Added the `--prod` flag to `yarn install` happening inside shell apps so Yarn installs only the real dependencies of the project. This omits dependencies like:
- `babel-core`, `prettier`, `eslint` — we won't transpile, analyze, prettify JS when building the project,
- `expo-cli` — we could use `expo-cli` when building the project, but we don't (`expotools` are used and they are installed in the shell app)
- `expo-yarn-workspaces` — the `--ignore-scripts` removes the need for `expo-yarn-workspaces`

### Test plan

I have tested these changes as part of https://github.com/expo/turtle/pull/209. I have built two apps on SDK37 and both were built successfully.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+
+- Turtle no longer installs all dependencies of shell apps projects — only the production dependencies are installed now.
```
Comment (wkozyra95) [inline CHANGELOG.md:12]: changelog is about turtle-cli, so unrelated changes shouldn't be present here

**Comment by sjchmiela** [comment]
Added `--production` flag, as per:

```
       o npm install (in package directory, no arguments):
           Install the dependencies in the local node_modules folder.
           In global mode (ie, with -g or --global appended to the command),
           it installs the current package context (ie, the current working
           directory) as a global package.
           By default, npm install will install all modules listed as dependencies
           in npm help package.json.
 -->       With the --production flag (or when the NODE_ENV environment variable
           is set to production), npm will not install modules listed in
           devDependencies. To install all modules listed in both dependencies
           and devDependencies when NODE_ENV environment variable is set to production,
           you can use --production=false.  NOTE: The --production  flag  has  no  particular  meaning  when
         adding a
           dependency to a project.
```

```
  Usage: yarn install [flags]

  Yarn install is used to install all dependencies for a project.

  Options:

    ...
    --prod, --production [prod]         
```
